src/main/dialogs/basic/dialog.js

```diff
@@ -345,6 +391,96 @@ gpii.app.dialog.positionOnInit = function (that) {
+ * @param {Number} scaleFactor - The new scale factor to be applied.
+ * @param {Number} oldScaleFactor - The previous scale factor.
+ * @return {Number} The new width of the `BrowserWindow`.
+ */
+gpii.app.dialog.getScaledWidth = function (that, scaleFactor, oldScaleFactor) {
+    return scaleFactor * that.model.width / oldScaleFactor;
+};
+
+/**
+ * Given the new and the previous `scaleFactor` this function computes the new
+ * height of the component's `BrowserWindow`. In the typical case, the new height
+ * will be obtained by applying the ratio of the new and the previous `scaleFactor`
+ * to the current height but under some circumstances this may differ. For example,
+ * the height of the QSS widget is different depending on the setting which is
+ * represented in it.
+ * @param {Component} that - The `gpii.app.dialog` instance.
+ * @param {Number} scaleFactor - The new scale factor to be applied.
+ * @param {Number} oldScaleFactor - The previous scale factor.
+ * @return {Number} The new height of the `BrowserWindow`.
+ */
+gpii.app.dialog.getScaledHeight = function (that, scaleFactor, oldScaleFactor) {
+    return scaleFactor * that.model.height / oldScaleFactor;
+};
+
+/**
+ * Given the new and the previous `scaleFactor` this function computes the new
+ * offset of the component's `BrowserWindow` with respect to the lower right corner
+ * of the main screen. In the typical case, the new offset will be obtained by
+ * applying the ratio of the new and the previous `scaleFactor` to the current offset
+ * but under some circumstances this may differ. For example, the y-offset of the PSP
+ * should always be equal to the height of the QSS.
+ * @param {Component} that - The `gpii.app.dialog` instance.
+ * @param {Number} scaleFactor - The new scale factor to be applied.
+ * @param {Number} oldScaleFactor - The previous scale factor.
+ * @return {Object} The new offset of the `BrowserWindow`.
+ */
+gpii.app.dialog.getScaledOffset = function (that, scaleFactor, oldScaleFactor) {
+    return {
+        x: scaleFactor * that.model.offset.x / oldScaleFactor,
+        y: scaleFactor * that.model.offset.y / oldScaleFactor
+    };
+};
+
+/**
+ * When the `scaleFactor` changes, this function takes care of adjusting the
+ * dimensions and position of the dialog as well as of applying the new scale
+ * factor to the contents of the `BrowserWindow`.
+ * @param {Component} that - The `gpii.app.dialog` instance.
+ * @param {Number} scaleFactor - The new scale factor to be applied.
+ * @param {Number} oldScaleFactor - The previous scale factor.
+ */
+gpii.app.dialog.rescaleDialog = function (that, scaleFactor, oldScaleFactor) {
+    scaleFactor = scaleFactor || 1;
+    oldScaleFactor = oldScaleFactor || 1;
+
+    var width = that.getScaledWidth(scaleFactor, oldScaleFactor),
+        height = that.getScaledHeight(scaleFactor, oldScaleFactor),
+        offset = that.getScaledOffset(scaleFactor, oldScaleFactor);
+
+    that.applier.change("width", width);
```

Comment:
It might help to avoid some kinds of jank if all these changes are made simultaneously?
Either that.applier.change("", {width: width, height: height, offset: offset}) or else wrap this in a transaction